code/filealerter/pcapdecoder.q

```diff
@@ -0,0 +1,98 @@
+// Decoder is designed for pcap version 2.4
+// Info on pcap structure => https://www.kroosec.com/2012/10/a-look-at-pcap-file-format.html
+
+\d .pcap
+
+// size of headers in bytes and dict of protocol code conversions
+globheader: 24;
+packetheader: 16;
+
+// mapping from protocol id numbers to protocol name
+// add desired protocols to dictionary for decoder to be able to translate their id's
+allcodes:(enlist 6)!(enlist `TCP);
+
+
+// returns table of packet data
+buildtable:{[file]
+ // check that version number of input file is 2.4
+ pcapversioncheck: all 2 0 4 0 = "h"$4#4_read1 file;
```

Comment:
`read1` can read a specific number of bytes from a specific location e.g.

```q
read1(file;4;4)
```

Probably a little more efficient than reading the whole file into mem & dropping/taking

BTW I think you can also skip the `"h"$`, q can compare between different datatypes:

```q
q)2 0 4 0=0x02000400
1111b
```